client/taskenv/env.go

```diff
@@ -726,6 +815,34 @@ func (b *Builder) SetTaskLocalDir(dir string) *Builder {
 	return b
 }
 
+func (b *Builder) SetClientSharedAllocDir(dir string) *Builder {
+	b.mu.Lock()
+	b.clientSharedAllocDir = dir
+	b.mu.Unlock()
+	return b
+}
+
+func (b *Builder) SetClientTaskRoot(dir string) *Builder {
+	b.mu.Lock()
+	b.clientTaskRoot = dir
+	b.mu.Unlock()
+	return b
+}
+
+func (b *Builder) SetClientTaskLocalDir(dir string) *Builder {
+	b.mu.Lock()
+	b.clientTaskLocalDir = dir
+	b.mu.Unlock()
+	return b
+}
+
+func (b *Builder) SetClientTaskSecretsDir(dir string) *Builder {
+	b.mu.Lock()
+	b.clientTaskSecretsDir = dir
+	b.mu.Unlock()
+	return b
+}
```

Comment:
Since these are reqiured for `Build()`'s proper function it's a shame we can't enforce those dependencies through the New func or similar, but I can see how that gets awkward given the way Builder's are built up vs where FSIsolation is handled. A problem for another time perhaps.